Guidelines.md

```diff
@@ -597,6 +598,51 @@ An API should never trust the input from the client and input SHOULD be validatd
 - Consider logging input validation failures. Assume that someone who is performing hundreds of failed input validations per second is up to no good.
 - Use a secure parser for parsing the incoming messages. If you are using XML, make sure to use a parser that is not vulnerable to XXE and similar attacks.
 
+### 7.14 Logging and audit
+Services compliant with the Digipolis REST API Guidelines MUST write logging to `stdout` and `stderr` according to the [12factor][12factor-logging] guidelines. The format of the logs MUST follow the LogStash JSON standard, one entry per line without zonder `[, ]` or `,`. This is also communicated through the FAST procedure.
+
+```json
+{"@timestamp":"2018-02-22T13:23:51.625+00:00","@version":1,"message":"A creation event has been sent to the queue for sms entity with id 38","logger_name":"gent.digipolis.servicefactory.notification.sms.api.module.sms.service.impl.QueueSenderImpl","thread_name":"http-nio-8080-exec-1","level":"INFO","level_value":20000,"X-Correlation-Id":"551aca4a-1b04-45eb-b4f4-e465470ce40b"}
+{"@timestamp":"2018-02-22T13:23:51.628+00:00","@version":1,"message":"Response: [method=POST,uri=/v1/sms,query=<null>,headers={Location=[/v1/sms/38]},body=[locations=[/v1/sms/38]],statusCode=201,responseTime=15]","logger_name":"gent.digipolis.servicefactory.notification.sms.api.module.sms.rest.SmsController","thread_name":"http-nio-8080-exec-1","level":"INFO","level_value":20000,"X-Correlation-Id":"551aca4a-1b04-45eb-b4f4-e465470ce40b"}
+```
+
+A JSON object should at least have following fields:
+
+| Field            | Description
+|------------------|------------
+| @timestamp       | Time of the log event. (yyyy-MM-dd'T'HH:mm:ss.SSSZZ) 
+| @version         | Logstash format version (e.g. 1) 
+| message          | Formatted log message of the event
+| logger_name      | Name of the logger that logged the event
+| X-Correlation-Id | Correlation ID header of the request
```

Comment:
X- is usually used in HTTP headers to denote it's a custom header. Do we need that in the name for the field here too?

Reply:
Taken from the FAST procedure spec, asked Jan.

Reply:
I feel we should check this, because I'm 99% sure this has been decided by non-technical people.